File: server/src/models/store_entry.rs
use serde::{Deserialize, Serialize};

/// A document that represents user ownership of a title in a storefront.
#[derive(Serialize, Deserialize, Default, Debug)]
pub struct StoreEntry {
    pub id: i64,
    pub title: String,
    pub storefront_name: String,

    #[serde(skip_serializing_if = "String::is_empty")]
    pub url: String,

    #[serde(skip_serializing_if = "String::is_empty")]
    pub image: String,
}

File: client/lib/modules/documents/store_entry.dart
class StoreEntry {
  final String id;
  final String title;
  final String storefront;

  final String? url;
  final String? image;

  StoreEntry({
    required this.id,
    required this.title,
    required this.storefront,
    this.url,
    this.image,
  });

  StoreEntry.fromJson(Map<String, dynamic> json)
      : this(
          id: json['id']!,
          title: json['title']!,
          storefront: json['storefront_name']!,
          url: json['url'],
          image: json['image'],
        );

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'storefront_name': storefront,
      if (url != null) 'url': url,
      if (image != null) 'image': image,
    };
  }
}
